### ISSUE TITLE
Task Dependencies

### ISSUE THREAD

**Issue by webdevsoup** [loduis/TeamWorkPmPhpApi#36]
Is there a way to set Task Dependencies via the API?  I have set up to use "predecessors", but that doesn't seem to set the dependency.  Any thoughts?

**Comment by simonschaufi** [comment]
If you could give me a code snippet what you are trying to do, I could help you maybe.

**Comment by webdevsoup** [comment]
```
// If we have tasks for our task list, foreach through them
if (!empty($list)) {
            foreach ($list as $content => $item) {
                // Assign our tags
                $tags = $item['tags'];
                // If we have predecessors, build them here
                if(!empty($item['predecessor'])) {
                    $predecessors = array("id"=>$subTaskReference[$item['predecessor']], "type"=>"complete");
                } else {
                    $predecessors = array();
                }
               // Encode our predecessors
                if(!empty($predecessors)) {
                    $predecessors = json_encode($predecessors);
                }
                // create the task
                $task = \TeamWorkPm\Factory::build('task');
                $task_id = $task->save([
                    'task_list_id' => $task_list_id,
                    'content' => $content,
                    'start_date' => date('Ymd'),
                    'private' => 1,
                    'tags' => $tags,
                    'predecessors' => "[".$predecessors."]"
                ]);
                $subTaskReference[$item['int_id']] = $task_id;
            }
        }
```

**Comment by simonschaufi** [comment]
Probably the field is missing here: https://github.com/loduis/TeamWorkPmPhpApi/blob/master/src/Task.php#L7 
Could you please play with it until it works and make a pull request? I'm able to merge it.

**Comment by webdevsoup** [comment]
I've actually already put it there.  It's still not setting them.

**Comment by simonschaufi** [comment]
I just saw that someone else had the same issue: https://github.com/loduis/TeamWorkPmPhpApi/pull/22
Can you test if that works for you?

**Comment by webdevsoup** [comment]
I followed the solution you posted here, and it did not work for me.  Still no dependencies assigned.

**Comment by simonschaufi** [comment]
Maybe the format you are assigning is wrong? What if you get the data of a task?

**Comment by webdevsoup** [comment]
According to the documentation inside of Teamwork, it's required to have a format type of array, with a minimum of an key of "id" and a key of "type".  So, that's what I'm assigning it.  

As far as getting the data of a task, hold on.  Let me create one, and see what it comes back with.

**Comment by webdevsoup** [comment]
If I assign it within Teamwork itself, and then get the data, it is there:
[predecessors] => ArrayObject Object ( [storage:ArrayObject:private] => Array ( [0] => ArrayObject Object ( [storage:ArrayObject:private] => Array ( [parentTaskId] => [name] => [Strategist] Day 8: Send Creative to Client for Approval [id] => 8612517 [type] => start ) ) )

**Comment by webdevsoup** [comment]
Alright, so I made a few changes, and it's all set now.  It's taking my predecessors.

Predecessors need to be put in to a mutli-dimensional array, such as this:

`$predecessors = array(array("id"=>$subTaskReference[$item['predecessor']], "type"=>"complete"));`

Once I had them like that, they took just fine.

**Comment by webdevsoup** [comment]
I used the solution to pull request #22

**Comment by simonschaufi** [comment]
great, so I can merge it?

**Comment by webdevsoup** [comment]
Yessir.

**Comment by webdevsoup** [comment]
Closing, fixed in pull request #22
